client/src/pages/MainPage/MainPage.tsx

```diff
@@ -2,45 +2,47 @@ import { useNavigate } from "react-router-dom";
 import Header from "../../components/PageComponents/Header/Header";
 import Footer from "../../components/PageComponents/Footer/Footer";
 import ListContainer from "./ListContainer/ListContainer";
-import { datePick,
-  linkCalendarPage,
-  locationTextArea,
-  pageContainer,
-  textArea,
-  weatherText
-} from "./MainPage.css";
+import { datePick, linkCalendarPage, locationTextArea, pageContainer, textArea, weatherText } from "./MainPage.css";
 import { appContainer } from "../../App.css";
+import useModalStore, { toggleLocationModal } from "../../store/store";
+import LocationModal from "../../components/ModalComponents/LocationModal/LocationModal";
 
 const MainPage = () => {
+  const locationModal = useModalStore((state) => state.locationModal);
+
   const navigate = useNavigate();
 
   const gotoCalendarPage = () => {
     navigate("/calendar");
   };
 
   return (
-    <div className={appContainer}>
-      <Header />
-      <div className={pageContainer}>
-        <div className={locationTextArea}>현재 지역은 location 입니다.</div>
-        <div>
-          <ListContainer />
-        </div>
-        <div className={textArea}>
-          <div className={datePick}>
-            + 월별 일정 추가
+    <>
+      {locationModal ? <LocationModal /> : null}
```

Comment:
이렇게 표현해줘도 괜찮을 것 같네요!
```suggestion
      {locationModal && <LocationModal />}
```